src/main/java/ch/njol/skript/events/EvtServerTime.java

```diff
@@ -0,0 +1,147 @@
+				.examples(
+					"on server time of 14:20:",
+					"on real time at 2:30am:",
+					"on server time at 6:10 pm:",
+					"on real time of 5:00 am and 5:00 pm:",
+					"on server time of 5:00 and 17:00:"
+				)
+				.since("INSERT VERSION");
+
+		timer = new Timer("EvtServerTime-Tasks");
+	}
+
+	private Literal<Time> times;
+	private boolean unloaded = false;
+	private List<ServerTimeInfo> infoList = new ArrayList<>();
+
+	@Override
+	public boolean init(Literal<?>[] args, int matchedPattern, ParseResult parseResult) {
+		//noinspection unchecked
+		times = (Literal<Time>) args[0];
+		return true;
+	}
+
+	@Override
+	public boolean postLoad() {
+		Calendar currentCalendar = Calendar.getInstance();
+		currentCalendar.setTimeZone(TimeZone.getDefault());
+		for (Time time : times.getArray()) {
+			Calendar expectedCalendar = Calendar.getInstance();
+			expectedCalendar.setTimeZone(TimeZone.getDefault());
+			expectedCalendar.set(Calendar.MINUTE, time.getMinute());
+			expectedCalendar.set(Calendar.SECOND, 0);
+			expectedCalendar.set(Calendar.MILLISECOND, 0);
+			expectedCalendar.set(Calendar.HOUR_OF_DAY, time.getHour());
+			// Ensure the execution time is in the future and not the past
+			while (expectedCalendar.before(currentCalendar)) {
+				expectedCalendar.add(Calendar.HOUR_OF_DAY, 24);
+			}
+			ServerTimeInfo info = new ServerTimeInfo(time, expectedCalendar.getTimeInMillis());
+			infoList.add(info);
+			createNewTask(info);
+		}
+		return true;
+	}
+
+	@Override
+	public void unload() {
+		unloaded = true;
+		for (ServerTimeInfo info : infoList) {
+			if (info.task != null)
+				info.task.cancel();
+		}
+		timer.purge();
+	}
+
+	@Override
+	public boolean check(Event event) {
+		throw new UnsupportedOperationException();
+	}
+
+	@Override
+	public boolean isEventPrioritySupported() {
+		return false;
+	}
+
+	@Override
+	public String toString(@Nullable Event event, boolean debug) {
+		return "server time of " + times.toString(event, debug);
+	}
+
+	private void execute() {
+		ServerTimeEvent event = new ServerTimeEvent();
+		SkriptEventHandler.logEventStart(event);
+		SkriptEventHandler.logTriggerStart(trigger);
+		trigger.execute(event);
+		SkriptEventHandler.logTriggerEnd(trigger);
+		SkriptEventHandler.logEventEnd();
+	}
+
+	private void preExecute(ServerTimeInfo info) {
+		// Safety check, ensure this 'EvtServerTime' was not unloaded
+		if (unloaded)
+			return;
+		// Bump the next execution time by the appropriate amount
+		info.executionTime += HOUR_24_MILLISECONDS;
+		// Reschedule task for new executionTime
+		createNewTask(info);
+		// Activate trigger
+		execute();
+	}
+
+	private void createNewTask(ServerTimeInfo info) {
+		TimerTask task = new TimerTask() {
+			@Override
+			public void run() {
+				preExecute(info);
+			}
+		};
+		info.task = task;
+		timer.schedule(task, new Date(info.executionTime));
+	}
+
+	private static class ServerTimeInfo {
```

Comment:
record

Reply:
Can not use record, `task` has to be changed.